vendor/modules.txt

```diff
@@ -283,12 +288,20 @@ github.com/onsi/gomega/matchers/support/goraph/edge
 github.com/onsi/gomega/matchers/support/goraph/node
 github.com/onsi/gomega/matchers/support/goraph/util
 github.com/onsi/gomega/types
-# github.com/opencontainers/runc v1.0.0 => github.com/opencontainers/runc v1.0.0-rc92
+# github.com/opencontainers/runc v1.0.0
 ## explicit
 github.com/opencontainers/runc/libcontainer/cgroups
 github.com/opencontainers/runc/libcontainer/cgroups/devices
+github.com/opencontainers/runc/libcontainer/cgroups/ebpf
+github.com/opencontainers/runc/libcontainer/cgroups/ebpf/devicefilter
```

Comment:
I believe this is for the disk hotplug. In cgroupsv1 we could just write to the .allow and .deny files, but in v2 you need an ebpf that manages the access.

Reply:
Yes, exactly. As Alex said, cgroups v2 deals with devices subsystem completely differently using eBPFs instead of file system interface.

Also, bare in mind that runc made a huge update to their code, which caused many files in under vendor/ to change.

Reply:
Are we adding our program there, or are we replacing an existing one? With other words, can we by accident loose some policy decisions instead of just adding our allows/denies? I could not look close enough to judge that.

Reply:
So for hotplug to allow access to block devices that we just added to the node using the attachment pod, we have to locate the container the virt-handler is using, and then modify the allowed devices to allow access to the new block device. For cgroupsv1 this is done using the devices.list (which we modify by writing to devices.allow and devices.deny to remove). For cgroupsv2, this mechanism is replaced by the ebpf which governs the access to the block device. Itamar abstracted this difference away so from a hotplug perspective we are just calling a function to allow/deny access and in the background based on which cgroup version we are running it uses either the devices.list or the ebpf.

Reply:
Hey @rmohr,
Currently there is no way of knowing the currently defined rules with eBPFs as they are represented with bytecode instead of text-based policy.

To overcome this I've introduced a small mechanism to save the past rules and override only whatever is requested explicitly so that we avoid overriding. You can see the relevant code [here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/cgroup/util.go#L56).

Also, I've created an [Issue for runc](https://github.com/opencontainers/runc/issues/3196) since I think this should be part of their code, not our's. The issue is ongoing.